### PR TITLE
Created MDReg() class

### PR DESCRIPTION
**Pull request by plaresmedima**
Need to pause this development for now so committing where I am.

All the examples in example.py are working on the example DICOM dataset. 

The README is extended but *not all functionality described* is currently implemented. It is written out already in the README to create specifications for development. These are smaller changes related to package structure and API.

### CODE REVIEW

**Comment by JSousa-UoL** [comment]
Hi @plaresmedima  thank you for your Pull Request. There are 2 points to discuss here.

1. Conflicts: There are two conflicts in the `examples.py` file. These are my fault because after the previous PR you submitted I did 2 direct commits to the main branch with small changes to the `examples.py`. The main change was adding a check if there are dicom files in the given data folder and print a warning if the folder is empty. I can resolve these conflicts easily by clicking on the "Resolve conflicts" button above and create a new commit for this branch.

Capture

2. There's a big and important implication with this PR. The MDR being implemented in the iBEAt and UKRIN-MAPS pipelines is still the old one. We will need to be extremely careful when determining the `mdr-library` version in those analysis pipelines. The other point is that the suggested current folder structure isn't friendly with the `pip install`. This works as a local repository, but having the MDReg() class in `main.py` doesn't allow you to have that class pip installable.

Continuing point 2.
In order to have the MDReg() class correctly available, you need to have it in a folder first. The big question is: How do you wish other people to import MDReg()? Surely they won't be writing "from main import MDReg". Assuming what you want is "from module_x.script_y import MDReg, then you would have folder "module_x" containing the file "script_y" where you can find the MDReg() class. I'm sharing an email you sent earlier this year and then we can make a decision based on it.

'''
I'm not sure about the package structure because mdr is the same thing as mdreg, and then you also have model_driven_registration as a function.

so to use it you would have to write

from mdreg.mdr import model_driven_registration

That's 3 times mdr.. 

And then all the folders like models etc are also distributed as part of the package I assume? 

Maybe something like:

mdreg.main (one public function which can be called coreg for instance instead of model_drive_registratioN)
mdreg.tools
mdreg.models
mdreg.examples (a simplification of what is now tests)
mdreg.elastixpars (examples of elastix parameter files)
mdreg.data (some demo data for tutorials)
mdreg,docs
'''

**Comment by plaresmedima** [comment]
Thanks @JSousa-UoL for the review. 

I didn't add an __init__ file yet as I was worried about conflicts but note in main.py the top line is

__all__ = ['MDReg']

so if you create an __init__ file in the top folder MDR_Library then that __init__ file should state

from MDR_Library.main import *

this then exposes MDReg to external callers, so you should be able to do

from mdr_library import MDReg

Do I get this right?

Re naming I would prefer we rename 

mdr_library -> mdreg

**Comment by JSousa-UoL** [comment]
I see what the idea is. I think what confuses you is the fact that the Github repository is named "MDR_Library" but that has absolutely nothing to do with the "pip install mdr-library"

In that case, if your final goal is the user to call "from mdreg import MDReg", then create a folder named `mdreg` and move your current `main.py` file to that folder and create the `__init__.py` as you're planning. If you wish to call "from mdreg import models_signal.T2" then you should move the `models_signal` folder into the newly created `mdreg` folder.

After all this changes, what we can do is to leave the "mdr-library" as it is and I can create a new PYPI package named `mdreg`. It's actually better this way, because that means none of these new changes will go to `mdr-library` and `mdr-library` stays as it is. I understand if it's difficult to understand, you can always arrange a call. I'll try to invite you to PYPI so you can get familiar with this pip install dynamics.

**Comment by plaresmedima** [comment]
Yes that makes total sense @JSousa-UoL Joao - indeed this is the idea

from mdreg import MDReg
from mdreg.models import T2star

etc

I will rearrange the folder structure following your suggestion and make a new PR.

Thanks a lot!

**Comment by JSousa-UoL** [comment]
Great @plaresmedima! 👍 

No need to create a new PR, you can just keep committing to this branch and I'll approve in the end when we're happy with the new structure.

File: results/README.md [hunk 1]
```diff
@@ -1 +1 @@
-Placeholder for test results
\ No newline at end of file
+Placeholder for test data
```
Comment (JSousa-UoL) [inline results/README.md:1]: Maybe "Placeholder for test data results"?

File: examples.py [hunk 2]
```diff
@@ -16,488 +16,317 @@
...
-import models_signal.T2star_simple as T2star_simple
-
-# To read and write from/to other locations, change these path names
 data = os.path.join(os.path.dirname(__file__), 'data')
 results = os.path.join(os.path.dirname(__file__), 'results')
-# Check if `data` folder has more files besides README.md
-if len([files for path, subdirs, files in os.walk(data)]) <= 1: warnings.warn("Data folder is empty!")
+elastix_pars = os.path.join(os.path.dirname(__file__), 'elastix')
 
+# To read and write from/to other locations, set these path names
```
Comment (JSousa-UoL) [inline examples.py:31]: Just a warning that these are your local paths and although it's ok for your local tests, it's not the best way to share examples publicly. Maybe also considering adding a check of folder `data` being empty or not when running these examples.

Something like
```
# Check if `data` folder has more files besides README.md
if len([files for path, subdirs, files in os.walk(data)]) <= 1: warnings.warn("Data folder is empty!")
```

**Comment by JSousa-UoL** [comment]
Thank you for the recent commit and for merging @plaresmedima. Yesterday I created a branch from the main branch named "mdr-library" so that we can keep a copy of the "old" MDR_Library in case we need to recover it one day in the future. Before I close this Pull Request, I'll list my next actions after closure:

1. Rename the Github repository to "mdreg" and propagate that change across the folders and documentation.
2. Create a new PYPI package named `mdreg`. Although the Github name "MDR_Library" won't exist anymore, we will keep a branch for that and the `mdr-library` pip install package will still exist and be available and functional. Then, it'll be up to us to delete it in the future when it becomes 100% deprecated.
3. Create a new Pull Request with changes in the `setup.py` file so that the new `mdreg` pip install package is generated and uploaded to PYPI for the first time.

**Pull request merged**